cli/src/config.rs

```diff
@@ -85,6 +87,31 @@ impl Manifest {
         }
     }
 
+    pub fn check_crate_version(&self, program: &str, name: &str) -> Result<(), semver::Error> {
+        if let Some(dep) = self.dependencies.get(name) {
+            let dep_version = match dep {
+                Dependency::Simple(v) => v.clone(),
+                Dependency::Detailed(d) => {
+                    if let Some(v) = d.version.clone() {
+                        v
+                    } else {
+                        return Ok(());
```

Comment:
I feel like a warning would be helpful here as well. sth like `Warning: No version selected for dependency "anchor-lang"`

Reply:
True, the only justifiable case for it _not_ being versioned is internal to the repository when everything is path based. I'll add one there as well, good call.

Reply:
done!